android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -111,6 +116,44 @@ class TranslationsMiddleware(
         }
     }
 
+    /**
+     * Retrieves the list of never translate sites using [scope] and dispatches the result to the
+     * store via [TranslationsAction.SetNeverTranslateSitesAction] or else dispatches the failure
+     * [TranslationsAction.TranslateExceptionAction].
+     *
+     * @param context Context to use to dispatch to the store.
+     * @param tabId Tab ID associated with the request.
+     */
+    private suspend fun getNeverTranslateSites(
+        context: MiddlewareContext<BrowserState, BrowserAction>,
+        tabId: String,
+    ) = withContext(Dispatchers.IO) {
```

Comment:
We recently explored more on this `Dispatchers.IO` over in https://github.com/mozilla-mobile/firefox-android/pull/5409 and a change landed today. Looks like the best recommended path forward is no `withContext` and dispatch once with `scope.launch`.

```suggestion
private suspend fun getNeverTranslateSites(
        context: MiddlewareContext<BrowserState, BrowserAction>,
        tabId: String,
        ) {
        engine.getNeverTranslateSiteList(
```

Reply:
Gotcha, removed `scope.launch`